integration/src/test/java/org/mule/test/construct/FlowRefTestCase.java

```diff
@@ -386,6 +386,14 @@ public void dynamicFlowRefWithSdkOperation() throws Exception {
     flowRunner("dynamicFlowRefWithSdkOperation").run();
   }
 
+  @Test
+  @Issue("MULE-19319")
+  @Description("For each with a flow ref and max concurrency finish processing")
+  public void forEachWithFlowRefAndMaxConcurrency() throws Exception {
+    flowRunner("foreachWithFlowRefAndMaxConcurrency").dispatchAsync(asyncFlowRunnerScheduler);
```

Comment:
why do you need to run this async? doesn't this test timeout already with the bug?